resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetDeltaJob.java

```diff
@@ -236,6 +236,12 @@ public void setDisabled(boolean disabled) {
 	}
 
 	public void shutdown() {
+		try {
+			cancel(); // otherwise next job runs into "already shutdown."
```

Comment:
Makes sense to cancel and join the job during shutdown to stop the job as fast as possible and avoid that the job still runs while the workspace is already shutdown. I would expect this to solve the "already shutdown" issue because _this_ job does not run into "already shutdown" anymore, correct?
So just for understanding: what do you mean by 'otherwise **next** job runs into "already shutdown."'?

Reply:
'this Job's next execution'

Reply:
thanks for clarification!